hicommonwealth/commonwealth
# Migrated Profiles to RQ

## Link to Issue
Closes: #4892, #4819

## Description of Changes
- Migrated `newProfile` controller APIs to react query and cleaned up unnecessary state/apis and logic
- Cleaned up `user.tsx` to be more descriptive and removed unnecessary logic

## "How We Fixed It"

Not a bug, just a migration. We migrated `newProfile` controller to react query.

## Test Plan
- Visit different pages that show user info (essentially the user.tsx) component and validate that correct data is displayed for each user
- Go to edit profile and make sure that editing your profile works perfectly and visiting subsequent pages after that work perfectly as well (without reloading the page)
- [Important] - Make sure that creating a new account works perfectly with multiple wallets
- [Important] - Make sure that logging into an existing account works perfectly and visiting subsequent pages after that work perfectly as well (without reloading the page)

## Deployment Plan
N/A

## Other Considerations
- The `newProfile` is not entirely cleaned up. I left some event emitters there intentionally as to not break the wallet/auth logic. We need to come up with plan to cleanup manual event emitters. @jnaviask @masvelio 
- The `newProfile` tests would be broken after this change, we can remove those tests.
